packages/core/src/components/Sequence/Sequence.tsx

```diff
@@ -0,0 +1,157 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { Fragment, FC, useState } from 'react';
+import {
+  Stepper,
+  Step,
+  StepContent,
+  StepLabel,
+  Typography,
+  makeStyles,
+} from '@material-ui/core';
+import SequenceFooter from './SequenceFooter';
+
+const useStyles = makeStyles((theme: any) => ({
+  content: {
+    padding: theme.spacing(0, 4, 4, 4),
+    backgroundColor: 'white',
+  },
+  completionBox: {
+    padding: theme.spacing(3),
+  },
+}));
+
+type Orientation = 'horizontal' | 'vertical';
+
+export type StepActions = {
+  showNext?: boolean;
+  canNext?: () => boolean;
+  onNext?: () => void;
+  nextStep?: (current: number, last: number) => number;
+  nextText?: string;
+
+  showBack?: boolean;
+  backText?: string;
+  onBack?: () => void;
+
+  showRestart?: boolean;
+  canRestart?: () => boolean;
+  onRestart?: () => void;
+  restartText?: string;
+
+  completedButton?: JSX.Element;
+};
+
+export type StepType = {
+  title: string;
+  actions?: StepActions;
+  content: JSX.Element;
+};
+
+export interface SequenceProps {
+  completed?: StepType;
+  elevation?: boolean;
+  orientation: Orientation;
+  onSequenceStepChange?: (prevIndex: number, nextIndex: number) => void;
+  steps: StepType[];
+  footerProps?: any;
+}
+
+const Sequence: FC<SequenceProps> = ({
+  completed,
+  elevation,
+  orientation,
+  onSequenceStepChange,
+  steps,
+  footerProps,
+}) => {
+  const classes = useStyles();
+  const [stepIndex, setStepIndex] = useState<number>(0);
+  const [stepArray, setStepArray] = useState<number[]>([0]);
+  const isVertical = orientation === 'vertical';
+  const horizontalStep =
+    !isVertical && steps[stepIndex] ? steps[stepIndex].content : false;
+
+  return (
+    <Fragment>
+      <Stepper
+        activeStep={stepIndex}
+        orientation={orientation}
+        elevation={elevation ? 2 : 0}
+      >
+        {steps.map((step, i) => (
+          <Step key={`step${i}`}>
+            <StepLabel>
+              <Typography variant="h6">{step.title}</Typography>
+            </StepLabel>
+            {isVertical && (
+              <StepContent>
+                {step.content}
+                <SequenceFooter
+                  stepIndex={i}
+                  setStepIndex={setStepIndex}
+                  stepArray={stepArray}
+                  setStepArray={setStepArray}
+                  length={steps.length}
+                  onSequenceStepChange={onSequenceStepChange}
+                  actions={step.actions || {}}
+                  {...footerProps}
+                />
+              </StepContent>
+            )}
+          </Step>
+        ))}
+      </Stepper>
+      {stepIndex === steps.length && !!completed && (
```

Comment:
I'd probably hint at what the check is by assigning to a const further up, like `isCompleted`